.github/workflows/build-git-installers.yml

```diff
@@ -101,34 +101,65 @@ jobs:
           git remote add -f origin https://github.com/git-for-windows/git &&
           git fetch "https://github.com/${{github.repository}}" refs/tags/${tag_name}:refs/tags/${tag_name} &&
           git reset --hard ${tag_name}
+      - name: Log in to Azure
+        uses: azure/login@v2
+        with:
+          client-id: ${{ secrets.AZURE_CLIENT_ID }}
+          tenant-id: ${{ secrets.AZURE_TENANT_ID }}
+          subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
       - name: Prepare home directory for code-signing
         env:
-          CODESIGN_P12: ${{secrets.CODESIGN_P12}}
-          CODESIGN_PASS: ${{secrets.CODESIGN_PASS}}
-        if: env.CODESIGN_P12 != '' && env.CODESIGN_PASS != ''
+          AZURE_VAULT: ${{ secrets.AZURE_VAULT }}
+          CODESIGN_CERT_SECRET_NAME: ${{ secrets.WIN_CODESIGN_CERT_SECRET_NAME }}
+          CODESIGN_PASS_SECRET_NAME: ${{ secrets.WIN_CODESIGN_PASS_SECRET_NAME }}
+        if: env.CODESIGN_CERT_SECRET_NAME != '' && env.CODESIGN_PASS_SECRET_NAME != ''
         shell: bash
         run: |
           cd home &&
           mkdir -p .sig &&
-          echo -n "$CODESIGN_P12" | tr % '\n' | base64 -d >.sig/codesign.p12 &&
-          echo -n "$CODESIGN_PASS" >.sig/codesign.pass
+          az keyvault secret download \
+            --vault-name "$AZURE_VAULT" \
+            --name "$CODESIGN_CERT_SECRET_NAME" \
+            --encoding base64
+            --file .sig/codesign.p12 &&
+          az keyvault secret download \
+            --vault-name "$AZURE_VAULT" \
+            --name "$CODESIGN_PASS_SECRET_NAME" \
+            --encoding base64
```

Comment:
The password is stored in ASCII, isn't it? If so, we probably want to use `show` instead of `download` and `--output tsv` instead of `--encoding base64`, and redirect the output to a file.

Reply:
Ah, you are correct that `--encoding base64` is wrong here, but we can still use `az keyvault secret download` rather than `show --output tsv --query value >file`.

> ```
> --encoding -e        : Encoding of the secret. By default, will look for the 'file-encoding' tag
>                        on the secret. Otherwise will assume 'utf-8'.  Allowed values: ascii,
>                        base64, hex, utf-16be, utf-16le, utf-8.
> ```

We can just update the encoding type to `ascii` rather than `base64`. (or should it be `utf8`?)

Reply:
I am not masochist enough to choose a non-ASCII passphrase any time soon, so `ascii` will do 😉